src/object-store/src/shared_realm.cpp

```diff
@@ -318,6 +320,13 @@ void Realm::verify_in_write() const
     }
 }
 
+void Realm::verify_open() const
+{
+    if (is_closed()) {
+        throw InvalidTransactionException("Can't read a Realm that was closed.");
```

Comment:
I agree this should be improved, though I think we should add `verify_open()` guards in the right places around `SharedRealm`. There are so many avenues through which `SharedRealm` is accessed, that it would be burdensome for a binding to methodically check `is_closed()` for every relevant method call. I'm going to be MIA, so I leave this PR up to you and @alazier to find the best solution. The test I added below will cause a crash unless this is resolved somehow. :smile: